### ISSUE TITLE
Gravitee | All services are running but cannot access API-ManagmentUI Portal

### ISSUE THREAD

**Issue by IvoIlic95** [gravitee-io/issues#3101]
Hello guys

I am still troubleshooting with the installation of Gravitee in Openshift. I managed to run all images crash free, this means that they are running. But I still can't come to the Portal UI where I can login into the Gravitee Tool.

Furthermore I am running Elasticsearch on a Microsoft Server with xpack disabled. This server is running smoothly without any errors. The MongoDB runs on a pod inside the Openshift Environment. The MongoDB has a Password, User assigned which is marked as XXXX inside my files that I will provide. The reason for that are Infineon Security Policies.

In terms of Memory I have following Information:

Resource Type | Used | Max
-- | -- | --
CPU (Request) | 600 millicores | 2 cores
Memory (Request) | 2 GiB | 4 GiB
CPU (Limit) | 1200 millicores | 2 cores
Memory (Limit) | 2 GiB | 4 GiB
Pods | 4 | 20

The memory never gets exhausted or even the cpu. There was never a memory lack when I am running my pods. 

I also tried to use this link that I have created. This link routes to the management-api:
http://management-api-gravitee.muc-gp.icp.infineon.com/management/

This link responded like this:
{
  "message" : "HTTP 404 Not Found",
  "http_status" : 404
}

http://management-api-gravitee.muc-gp.icp.infineon.com/management/apis/

This responded like this:

[ ]

http://management-api-gravitee.muc-gp.icp.infineon.com/management/portal/

This responded like this (many lines):

[Link_Reponse.txt](https://github.com/gravitee-io/issues/files/4071280/Link_Reponse.txt)

I tried so many options to get this Application running and sometimes I am reaching my goal faster and sometimes slower. At this moment I don't know what the problem here is. That's why I am gladly asking if someone of the Gravitee developers can help me out with this troubleshoot.

Here are my files that I mentioned previously. They contain every information that is hopefully needed. If not then you can freely ask for more information.

[Gateway_Info.txt](https://github.com/gravitee-io/issues/files/4071253/Gateway_Info.txt)
[ManagementApi_Info.txt](https://github.com/gravitee-io/issues/files/4071254/ManagementApi_Info.txt)
[ManagementUI_Info.txt](https://github.com/gravitee-io/issues/files/4071255/ManagementUI_Info.txt)

I hope that someone can help me to get this Application running. I am looking forward.

Kind regards,
Ivo Ilic

**Comment by brasseld** [comment]
Hi @IvoIlic95 

> But I still can't come to the Portal UI where I can login into the Gravitee Tool.

What is the issue exactly ?
Blank screen
Screen a red message
Not able to connect...

**Comment by IvoIlic95** [comment]
Hello

Well GraviteeIO is now working! I changed the port of my route and now I am able to enter the GraviteeIO Interface. 

Now I wanted to add a new user. This was not a problem at all but I also wanted to change the password of this user by pressing change password, which would send a link to his email so that he can reset it. The EMail never arrived because the EMail service of GRavitee is not enabled.

The log threw this message:
 io.gravitee.management.service.exceptions.EmailDisabledException: Email service is disabled

How do I enable the EMail service? Do I need some environment variables?

Kind regards,
Ivo Ilic

**Comment by brasseld** [comment]
Hi @IvoIlic95 

Check this guide: https://docs.gravitee.io/apim_installguide_management_api_configuration.html

**Comment by IvoIlic95** [comment]
Hello

Yeah I have seen this. But where do I have to add this infos? Are there any environment variables for the E-Mail configuration?

email:
  host: smtp.my.domain
  port: 465
  from: noreply@my.domain
  subject: "[Gravitee.io] %s"
  username: user@my.domain
  password: password

Are these the env vars?

Kind regards,
Ivo Ilic

**Comment by brasseld** [comment]
https://docs.gravitee.io/pages/apim/installation-guide/partial/how-to-configure.html#environment_variables

For example:
`gravitee_email_host=smtp.my.domain`

**Comment by IvoIlic95** [comment]
Ahhh okey okey, I am going to try this.
Is it okey to let you know if it works or not?

Kind regards,
Ivo Ilic

**Comment by brasseld** [comment]
> Is it okey to let you know if it works or not?

For sur, before closing the issue.

**Comment by IvoIlic95** [comment]
Sadly I have to wait for the mail server infos, my company needs some time to collect the information and pass it through the security.

Therefore, I have a another question. What does this info mean? 
[sync-1] INFO  i.g.m.s.d.DynamicPropertiesService - Dynamic properties service is disabled for: Forrest API [1.16]

-> This info appeard after adding a API to Gravitee.

Is this linked with the dashboard so that the infos from the api can be visualized? And how can I activate it?

Kind regards,
Ivo Ilic

**Comment by brasseld** [comment]
Ok,

It means that the API has no dynamic properties configured.
Perhaps it may be better to define it as DEBUG instead of INFO.

**Comment by IvoIlic95** [comment]
And where and how can I configure these dynamic properties?
Can I configure this inside the Gravitee application or should I add some specific environment variables?

**Comment by brasseld** [comment]
in API > properties (design view)

**Comment by IvoIlic95** [comment]
I can only find this in the Dashboard: Settings->Analytics->"ADD A NEW DASHBOARD API"

![image](https://user-images.githubusercontent.com/59047274/72724045-8f0cf600-3b82-11ea-9ef3-20b7b4d04737.png)

**Comment by brasseld** [comment]
API properties are not linked to analytics.

**Comment by IvoIlic95** [comment]
Then under Selected API->Design->Properties->New property?

Like in the picture below? And what should I type in (does it have a specific format?) under properties?

![image](https://user-images.githubusercontent.com/59047274/72727355-fd55b680-3b8a-11ea-8299-b33ab0086db6.png)

**Comment by brasseld** [comment]
If you don't need it, do not use it....

**Comment by IvoIlic95** [comment]
Good morning

Okey, and what about the analytics part. I have created a API by using a Swagger descriptor. Then I created a API plan and subscribed to it with my application. After that I clicked on the analytics button from the imported API. And it only shows me this:

![image](https://user-images.githubusercontent.com/59047274/72875672-124e5900-3cf5-11ea-8512-92958e7e401c.png)

Could it be that the API is doing nothing, and because of that it can't show me anything in the analytics panel or is there a another cause for it? And yes the API is getting requests and so on.

Kind regards,
Ivo Ilic
